experimental/Schemes/IdealSheaves.jl

```diff
@@ -247,35 +258,49 @@ end
 #  return IdealSheaf(X, C, new_dict)
 #end
 
-function +(I::IdealSheaf, J::IdealSheaf)
-  X = space(I)
-  X == space(J) || error("ideal sheaves are not defined over the same scheme")
-  new_dict = IdDict{AbsAffineScheme, Ideal}()
-  CI = default_covering(X)
-  for U in patches(CI)
-    new_dict[U] = I(U) + J(U)
-  end
-  return IdealSheaf(X, new_dict, check=false)
+function +(I::AbsIdealSheaf, J::AbsIdealSheaf)
+  return SumIdealSheaf([I, J])
 end
 
-function *(I::IdealSheaf, J::IdealSheaf)
-  X = space(I)
-  X == space(J) || error("ideal sheaves are not defined over the same scheme")
-  new_dict = IdDict{AbsAffineScheme, Ideal}()
-  CI = default_covering(X)
-  for U in patches(CI)
-    new_dict[U] = I(U) * J(U)
-  end
-  return IdealSheaf(X, new_dict, check=false)
+function +(I::SumIdealSheaf, J::AbsIdealSheaf)
+  return SumIdealSheaf(vcat(summands(I), [J]))
+end
+
+function +(I::AbsIdealSheaf, J::SumIdealSheaf)
+  return SumIdealSheaf(vcat([I], summands(J)))
+end
+
+function +(I::SumIdealSheaf, J::SumIdealSheaf)
+  return SumIdealSheaf(vcat(summands(I), summands(J)))
+end
```

Comment:
You could simplify this to
```suggestion
function +(I::AbsIdealSheaf, J::AbsIdealSheaf)
  return SumIdealSheaf(vcat(summands(I), summands(J)))
end
```
and similar for the `*` methods coming next if you simply add
```julia
summands(I::AbsIdealSheaf) = [I]
```